File: eeducation-front/src/app/model/Lesson.ts
import {Absence} from './Absence';
import {Comment} from './Comment';

export class Lesson {
  constructor(public id: number,
              public absences: Array<Absence>,
              public comments: Array<Comment>,
              public lessonContent: string,
              public name: string,
              public date: Date,
              public courseId: number) {
  }
}

File: eeducation/src/main/kotlin/rs/eeducation/model/Lesson.kt
package rs.eeducation.model

import java.util.*
import javax.persistence.*

@Entity
class Lesson(@Id
             @GeneratedValue
             var id: Long?,
             @ManyToOne
             var course: Course,
             @OneToMany
             var absences: Set<Absence>,
             @OneToMany
             var comments: Set<Comment>,
             @Column
             var name: String,
             @Column
             var date: Date,
             @Column
             var lessonContentId: String,
             @Column
             var linked: Boolean?
) {
}
